src/cursor/cursor.ts

```diff
@@ -403,52 +374,42 @@ export class Cursor extends CoreCursor {
   /**
    * Set a node.js specific cursor option
    *
-   * @function
-   * @param {string} field The cursor option to set ['numberOfRetries', 'tailableRetryInterval'].
-   * @param {object} value The field value.
-   * @throws {MongoError}
-   * @returns {Cursor}
+   * @param field The cursor option to set ['numberOfRetries', 'tailableRetryInterval' -.
+   *
+   * @param value - The field value.
    */
-  setCursorOption(field: string, value: object): Cursor {
+  setCursorOption(field: keyof CursorPrivate, value: CursorPrivate[keyof CursorPrivate]): this {
     if (this.s.state === CursorState.CLOSED || this.s.state === CursorState.OPEN || this.isDead()) {
-      throw MongoError.create({ message: 'Cursor is closed', driver: true });
+      throw new MongoError('Cursor is closed');
     }
 
-    if (fields.indexOf(field) === -1) {
-      throw MongoError.create({
-        message: `option ${field} is not a supported option ${fields}`,
-        driver: true
-      });
+    if (!FIELDS.includes(field)) {
+      throw new MongoError(`option ${field} is not a supported option ${FIELDS}`);
     }
 
-    this.s[field] = value;
-    if (field === 'numberOfRetries') this.s.currentNumberOfRetries = value;
+    Object.assign(this.s, { [field]: value });
+    if (field === 'numberOfRetries') this.s.currentNumberOfRetries = value as number;
     return this;
   }
 
   /**
    * Add a cursor flag to the cursor
    *
-   * @function
-   * @param {string} flag The flag to set, must be one of following ['tailable', 'oplogReplay', 'noCursorTimeout', 'awaitData', 'partial'].
-   * @param {boolean} value The flag boolean value.
-   * @throws {MongoError}
-   * @returns {Cursor}
+   * @param flag The flag to set, must be one of following ['tailable', 'oplogReplay', 'noCursorTimeout', 'awaitData', 'partial' -.
+   *
+   * @param value - The flag boolean value.
    */
-  addCursorFlag(flag: string, value: boolean): Cursor {
+  addCursorFlag(flag: string, value: boolean): this {
```

Comment:
If you define `FLAGS` as const up above, you can have a `CursorFlags` type which will be a union of all strings in the array.

```ts
const FLAGS = [
  'tailable',
  'oplogReplay',
  'noCursorTimeout',
  'awaitData',
  'exhaust',
  'partial'
] as const;
type CursorFlags = typeof FLAGS[number];
```

```suggestion
  addCursorFlag(flag: CursorFlags, value: boolean): this {
```